### PR TITLE
display error if token with same application name exists

### PR DESCRIPTION
**Pull request by phisch**
## Description
Since #27415 seems to be inactive, i created migrations for the authtoken table creation and unique constraint, catched the exception and rendered an appropriate error message.

## Related Issue
#24778
#27415 (PR)

## Motivation and Context
See related issues & pull requests.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: settings/js/panels/authtoken_view.js [hunk 1]
```diff
@@ -195,8 +195,12 @@
 				_this._newAppPassword.select();
 				_this._tokenName.val('');
 			});
-			$.when(creatingToken).fail(function() {
-				OC.Notification.showTemporary(t('core', 'Error while creating device token'));
+			$.when(creatingToken).fail(function(xhr) {
+				if (xhr.status === 409) {
+					OC.Notification.showTemporary(t('core', 'A token for that application already exists. Please choose another application name.'));
```
Comment (phisch) [inline settings/js/panels/authtoken_view.js]: @DeepDiver1975 can you get me the translations for that string?
Not sure if i will get translations by putting them into transifex myself (not under the owncloud organisation).

**Comment by phisch** [comment]
The added index is a problem because `name` is a text without a fixed length which then can't be used in a index. I imagine this is a text because it contains the useragent. Maybe we could set a fixed length (like 500) and cut Useragent strings that are too long?

**Comment by phisch** [comment]
Hm, that would also be a bad solution since it could run into unique constraint violations with old tokens...
It's quite bad that we use the same column for userdefined names and user-agent strings.
Will think of a propper solution once i have more time for it.

**Comment by PVince81** [comment]
@phisch close ?

I see one of the files is new migrations that you have added for the tokens table. However I think you already added migrations for that when working on the 9.0->10.0 upgrade. I suggest closing for now as this is not a critical issue and I don't think it's worth spending too much time on a low severity issue.

**Pull request closed without merging**